client/components/community-translator/index.jsx

```diff
@@ -0,0 +1,162 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import i18n, { localize } from 'i18n-calypso';
+import debugModule from 'debug';
+import { find, isEmpty } from 'lodash';
+/**
+ * Internal dependencies
+ */
+import Translatable from './translatable';
+import config from 'config';
+import User from 'lib/user';
+import userSettings from 'lib/user-settings';
+import { isCommunityTranslatorEnabled } from 'components/community-translator/utils';
+
+/**
+ * Local variables
+ */
+const debug = debugModule( 'calypso:community-translator' );
+const languages = config( 'languages' );
+const user = new User();
+
+class CommunityTranslator extends Component {
+	languageJson = null;
+	currentLocale = null;
+	initialized = false;
+
+	componentDidMount() {
+		this.setLanguage();
+
+		// wrap translations from i18n
+		i18n.registerTranslateHook( ( translation, options ) =>
+			this.wrapTranslation( options.original, translation, options )
+		);
+
+		// callback when translated component changes.
+		// the callback is overwritten by the translator on load/unload, so we're returning it within an anonymous function.
+		i18n.registerComponentUpdateHook( () => {} );
+		i18n.on( 'change', this.refresh );
+		user.on( 'change', this.refresh );
+		userSettings.on( 'change', this.refresh );
+	}
+
+	componentWillUnmount() {
+		i18n.off( 'change', this.refresh );
+		user.removeListener( 'change', this.refresh );
+		userSettings.removeListener( 'change', this.refresh );
+	}
+
+	setLanguage() {
+		this.languageJson = i18n.getLocale() || { '': {} };
+		const { localeSlug, localeVariant } = this.languageJson[ '' ];
```

Comment:
It's not obvious why we access the key `''` here. we could consider to add a comment to explain it here.